frontend/src/scenes/experiments/utils.ts

```diff
@@ -431,108 +451,150 @@ export function filterToExposureConfig(
-                math_hogql: entity.math_hogql,
-                properties: entity.properties,
+                kind: NodeKind.ExperimentFunnelMetricConfig,
+                funnel:
+                    events?.map(
+                        (event) =>
+                            ({
+                                kind: NodeKind.ExperimentFunnelStepConfig,
+                                event: event.id,
+                                order: event.order,
+                                properties: event.properties,
+                            } as ExperimentFunnelStepConfig)
+                    ) || [],
             }
-        } else if (entity.type === 'data_warehouse') {
-            return {
-                kind: NodeKind.ExperimentDataWarehouseMetricConfig,
-                name: entity.name,
-                table_name: entity.id,
-                timestamp_field: entity.timestamp_field,
-                events_join_key: entity.events_join_key,
-                data_warehouse_join_key: entity.data_warehouse_join_key,
-                math: entity.math || ExperimentMetricMathType.TotalCount,
-                math_property: entity.math_property,
-                math_hogql: entity.math_hogql,
+        case ExperimentMetricType.MEAN:
+            if (events?.[0]) {
+                return {
+                    kind: NodeKind.ExperimentEventMetricConfig,
+                    event: events[0].id as string,
+                    name: events[0].name,
+                    math: events[0].math || ExperimentMetricMathType.TotalCount,
+                    math_property: events[0].math_property,
+                    math_hogql: events[0].math_hogql,
+                    properties: events[0].properties,
+                }
+            } else if (actions?.[0]) {
+                return {
+                    kind: NodeKind.ExperimentActionMetricConfig,
+                    action: actions[0].id,
+                    name: actions[0].name,
+                    math: actions[0].math || ExperimentMetricMathType.TotalCount,
+                    math_property: actions[0].math_property,
+                    math_hogql: actions[0].math_hogql,
+                    properties: actions[0].properties,
+                }
+            } else if (data_warehouse?.[0]) {
+                return {
+                    kind: NodeKind.ExperimentDataWarehouseMetricConfig,
+                    name: data_warehouse[0].name,
+                    table_name: data_warehouse[0].id,
+                    timestamp_field: data_warehouse[0].timestamp_field,
+                    events_join_key: data_warehouse[0].events_join_key,
+                    data_warehouse_join_key: data_warehouse[0].data_warehouse_join_key,
+                    math: data_warehouse[0].math || ExperimentMetricMathType.TotalCount,
+                    math_property: data_warehouse[0].math_property,
+                    math_hogql: data_warehouse[0].math_hogql,
+                }
```

Comment:
Since this is the sister frunction to `metricToFilter` we could apply some of the same refactoring rules to reduce duplication.